src/mono/mono/mini/mini-amd64.c

```diff
@@ -674,6 +674,18 @@ add_valuetype (MonoMethodSignature *sig, ArgInfo *ainfo, MonoType *type,
 		return;
 	}
 
+	/* Can't use mini_class_is_simd () here as we don't have access to a MonoCompile */
+	if (m_class_is_simd_type (klass) && !sig->pinvoke && !is_return) {
+		if (*fr >= FLOAT_PARAM_REGS) {
+			pass_on_stack = TRUE;
+		} else {
+			ainfo->storage = ArgSIMDInSSEReg;
```

Comment:
Is it intentional this doesn't match the native ABI for the platform?

There are particular rules for when and how certain SIMD types are passed depending on the ABI of the underlying platform (Windows x64, Unix SysV, etc).

Reply:
This is just an experiment right now, it has many problems, including this one.